src/libs/antares/study-loader/include/antares/study-loader/IStudyLoader.h

```diff
@@ -0,0 +1,32 @@
+
+/*
+ * Copyright 2007-2024, RTE (https://www.rte-france.com)
+ * See AUTHORS.txt
+ * SPDX-License-Identifier: MPL-2.0
+ * This file is part of Antares-Simulator,
+ * Adequacy and Performance assessment for interconnected energy networks.
+ *
+ * Antares_Simulator is free software: you can redistribute it and/or modify
+ * it under the terms of the Mozilla Public Licence 2.0 as published by
+ * the Mozilla Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Antares_Simulator is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Mozilla Public Licence 2.0 for more details.
+ *
+ * You should have received a copy of the Mozilla Public Licence 2.0
+ * along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+ */
+
+#pragma once
+#include <antares/study/study.h>
+
+namespace Antares {
+class IStudyLoader {
+public:
+    virtual ~IStudyLoader() = default;
+    virtual std::shared_ptr<Antares::Data::Study> load() = 0;
```

Comment:
Indeed it was my intent. The difficulty is that the "default" implementation FileTreeStudyLoader poses some difficulties for this.
In FileTreeStudyLoader "load" encapsulates the current "application.prepare" behaviour that create and load the study. This part, and most notably "load" passe around a shared_ptr. Difficulties are:
- It is not possible to transforme a shared_ptr into a unique else it would have only been a matter of returning the proper type.
- application prepare stack use shared_ptr. Changing for unique would mean changing all usage. Not difficult, just a bit of a tedious task
- application.prepare stack and inner mechanisms are opaque. It's not just all function call, else we could have just pass a pointer to the study with study_.get(). There are some objects that retains ownership of the study during their lifetime, justifying the use of shared_ptr. The lifetime of said objects probably doesn't exceed prepare() but since it's "black box" I didn't want to edit the interfaces and behaviour too much.

I'm still open to discussion and would very much prefer being able to return a unique_ptr

Reply:
Ok let's keep it in mind for later days

Reply:
https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/2038